diff --git i/protos/rendezvous.proto w/protos/rendezvous.proto
index b2e5c01..a57919f 100644
--- i/protos/rendezvous.proto
+++ w/protos/rendezvous.proto
@@ -3,7 +3,7 @@ package hbb;
 
 message RegisterPeer {
   string id = 1;
-  int32 serial = 2;
+  int32 serial = 2;  string licence_key = 20; // Handover: admission token
 }
 
 enum ConnType {
@@ -89,7 +89,7 @@ message PunchHoleSent {
   NatType nat_type = 4;
   string version = 5;
   int32 upnp_port = 6;
-  bytes socket_addr_v6 = 7;
+  bytes socket_addr_v6 = 7;  string licence_key = 20; // Handover: admission token
 }
 
 message RegisterPk {
@@ -97,7 +97,7 @@ message RegisterPk {
   bytes uuid = 2;
   bytes pk = 3;
   string old_id = 4;
-  bool no_register_device = 5;
+  bool no_register_device = 5;  string licence_key = 20; // Handover: admission token
 }
 
 message RegisterPkResponse {
@@ -168,7 +168,7 @@ message RelayResponse {
   string version = 7;
   int32 feedback = 9;
   bytes socket_addr_v6 = 10;
-  int32 upnp_port = 11;
+  int32 upnp_port = 11;  string licence_key = 20; // Handover: admission token
 }
 
 message SoftwareUpdate { string url = 1; }
@@ -191,7 +191,7 @@ message LocalAddr {
   string relay_server = 3;
   string id = 4;
   string version = 5;
-  bytes socket_addr_v6 = 6;
+  bytes socket_addr_v6 = 6;  string licence_key = 20; // Handover: admission token
 }
 
 message PeerDiscovery {
